tests/bdd/minikube/setup.sh

```diff
@@ -0,0 +1,460 @@
+install_prerequisites() {
+  local -r os=${1}
+  local -r arch=${2}
+  local -r workspace=${3}
+
+  case "$os" in
+  "GNU/Linux")
+    case "$arch" in
+    "x86_64")
+      # Check if nix-shell prerequisites are present in PATH.
+      for bin in "${NIX_SHELL_PREREQUISITES[@]}"; do
+        must_exists_in_path "${bin}" >/dev/null
+      done
+
+      echo "Installing prerequisites for $os-$arch..."
+
+      # Install cri-dockerd, if not installed.
+      local cri_dockerd_error
+      which cri-dockerd &>/dev/null || cri_dockerd_error=$?
+      if [ -n "${cri_dockerd_error}" ]; then
+        echo "Downloading latest version of cri-dockerd..."
+
+        local -r url=$(github_asset_url_from_latest_release \
+          "Mirantis" "cri-dockerd" "^(cri-dockerd-[0-9]+.[0-9]+.[0-9]+.amd64.tgz)$")
+        github_asset_binary "${url}" "${workspace}" "cri-dockerd.tgz"
+        tar -xf "${workspace}"/cri-dockerd.tgz -C "${workspace}"
+        mkdir -p "$PREREQUISITES_BIN_DIR"
+        install -o root -g root -m 0755 "${workspace}"/cri-dockerd/cri-dockerd "$PREREQUISITES_BIN_DIR"
+
+        must_exists_in_path "cri-dockerd" "failed to install cri-dockerd" >/dev/null
+
+        # Cleanup entry for cri-dockerd binary.
+        prepend_cleanup_script "# Removes the cri-dockerd binary
+rm -f $PREREQUISITES_BIN_DIR/cri-dockerd"
+
+        echo "Downloaded latest version of cri-dockerd."
+      fi
+      # Check if cri-docker.socket systemd service is active. While this is extremely
+      # unlikely if cri-dockerd wasn't already installed, it is possible.
+      local cri_dockerd_service_error
+      systemctl is-active --quiet cri-docker.socket || cri_dockerd_service_error=$?
+      if [ -n "${cri_dockerd_service_error}" ]; then
+        # Download systemd service files
+        local -r cri_dockerd_version="v$(cri-dockerd --version 2>&1 | awk '{print $2}')"
+        local -r systemd_service_files_url="https://raw.githubusercontent.com/Mirantis/cri-dockerd/${cri_dockerd_version}/packaging/systemd"
+        cd "${workspace}"
+        ${CURL} "${systemd_service_files_url}"/cri-docker.service -o cri-docker.service
+        ${CURL} "${systemd_service_files_url}"/cri-docker.socket -o cri-docker.socket
+        cd - >/dev/null
+        install "${workspace}"/cri-docker.{service,socket} /etc/systemd/system
+        sed -i -e "s,/usr/bin/cri-dockerd,$PREREQUISITES_BIN_DIR/cri-dockerd," /etc/systemd/system/cri-docker.service
+        systemctl daemon-reload
+        systemctl enable --now --quiet cri-docker.socket
+        systemctl is-active --quiet cri-docker.socket || die "failed to set up cri-dockerd systemd service" 1
+
+        # Cleanup entry for cri-docker.socket systemd service
+        prepend_cleanup_script "# Disables and stops the systemd services
+systemctl disable --now --quiet cri-docker.socket
+systemctl disable --now --quiet cri-docker
+systemctl daemon-reload
+# Delete the service and configuration files for the systemd services
```

Comment:
all the bits from both scripts that deal with etc, maybe we should ensure they're not tried on read-only /etc/systemd directories?

Reply:
There's a `--skip-prerequisites` option to stop these things from starting. Apart from that, I plan to make the KVM-mode minikube the default one.

Reply:
Added a note for read-only /etc/systemd/system systems in the README.md